internal/secret/secret_test.go

```diff
@@ -4,67 +4,25 @@
 package secret
 
 import (
-	"github.com/microsoft/go-sqlcmd/internal/output"
-	"strings"
+	"github.com/microsoft/go-sqlcmd/internal/test"
 	"testing"
 )
 
-func TestEncryptAndDecrypt(t *testing.T) {
-	type args struct {
-		plainText string
-		encrypt   bool
-	}
-	tests := []struct {
-		name          string
-		args          args
-		wantPlainText string
-	}{
-		{
-			name:          "noEncrypt",
-			args:          args{"plainText", false},
-			wantPlainText: "plainText",
-		},
-		{
-			name:          "encrypt",
-			args:          args{"plainText", true},
-			wantPlainText: "plainText",
-		},
-		{
-			name:          "emptyStringForEncryptPanic",
-			args:          args{"", true},
-			wantPlainText: "",
-		},
-		{
-			name:          "emptyStringForDecryptPanic",
-			args:          args{"", true},
-			wantPlainText: "",
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+func TestEncodeAndDecode(t *testing.T) {
+	notEncrypted := Encode("plainText", false)
+	encrypted := Encode("plainText", true)
+	Decode(notEncrypted, false)
+	Decode(encrypted, true)
```

Comment:
shouldn't this assert the decoded string is correct?

Reply:
Agree, I have leaned to much on adding unit tests for the code coverage only, without doing enough validation.  I've started a new PR to ensure all the unit tests include validation, so we will have good code coverage with validating results as well.

I see the testify package also supports mocking, I will look at using the mocking functionality to speed up the unit tests around the install/uninstall functionality.